source/FreeRTOS_DNS.c

```diff
@@ -1248,17 +1248,6 @@
         /* Make sure all fields of the 'sockaddr' are cleared. */
         ( void ) memset( ( void * ) &xAddress, 0, sizeof( xAddress ) );
 
-        #if ( ipconfigUSE_IPv6 != 0 )
```

Comment:
Is this assignment causing any issue?

Reply:
You are referring to this code:

~~~c
#if ( ipconfigUSE_IPv6 != 0 )
    if( xFamily == ( BaseType_t ) FREERTOS_AF_INET6 )
    {
        xDNS_IP_Preference = xPreferenceIPv6;
    }
    else
    {
        xDNS_IP_Preference = xPreferenceIPv4;
    }
#endif /* ( ipconfigUSE_IPv6 != 0 ) */
~~~

The global variable `xDNS_IP_Preference` was introduced to indicate the preference for a DNS server: either IPv4 or IPv6.

When I test the DNS functions, I always try out 4 combinations: contact the server with either IPv4 or IPv6, and ask for either an A (IPv4) or an AAAA (IPv6) record.

What we could do it hide the variable `xDNS_IP_Preference` and add an access function.

In answer to your question: changing `xDNS_IP_Preference` disturbs the testing process. The variable should be under control of the DUT. At least, that is how it was designed.